tools/release_provenance/models.py

```diff
@@ -0,0 +1,61 @@
+import os
+from sqlalchemy import (
+    DateTime, Text, String, Column, ForeignKey, Table, UniqueConstraint, Integer )
+from sqlalchemy import create_engine
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import relationship, sessionmaker
+
+Base = declarative_base()
+
+def create_session():
+    # A connection string in postgresql:// style, for example: postgresql://user:pass@host/DB_NAME
+    connection = os.getenv("BUILD_DB_CONNECTION_STR")
+    if connection is None:
+        raise Exception("No BUILD_DB_CONNECTION_STR found, check model repository secrets")
+
+    engine = create_engine(connection)
+    Base.metadata.create_all(engine)
+
+    Session = sessionmaker(bind=engine, autoflush=False)
+    return Session()
+
+
+class ComponentBuild(Base):
+    __tablename__ = "component_build"
+
+    spack_hash = Column(String, primary_key=True, index=True)
+    spec = Column(String, nullable=False)
+    install_path = Column(String, nullable=False, unique=True)
```

Comment:
@aidanheerdegen would the install path be nessecarily unique? I suppose it's a similar issue to the spack_hash uniqueness problem.

Reply:
Given it's on a filesystem and two things named the same thing can't exist in the same location, and assuming we're not overwriting (which we shouldn't be), then yeah I think it should be unique.